### PR TITLE
Struct Returns

### PR DESCRIPTION
**Pull request by engineer124**
One of the rare instances where changing the number of arguments retains the same codegen. There's a pattern where a struct-returning functions takes an out-pointer as first arg and also return that pointer. Both seem to match but it seems likely that it was the first given the redundancy of having both an arg and a return give a pointer for a destination. See https://github.com/zeldaret/mm/pull/1466 and the conversation here https://discord.com/channels/688807550715560050/688851997352263820/1170905743566835836

This also revealed some more unused variables, which has `UNUSED` added to them

### CODE REVIEW

File: include/ultra64/xstdio.h
Comment (hensldm) [inline include/ultra64/xstdio.h:35]: formatter?

**Review by hensldm — APPROVED** [review]

File: include/ultra64/xstdio.h
Comment (Dragorn421) [reply include/ultra64/xstdio.h:35]: Yes
https://discord.com/channels/688807550715560050/688851317593997489/1170482092858155039
#1573

File: src/code/z_camera.c
Comment (Dragorn421) [inline src/code/z_camera.c:461]: We don't use unused in oot so far #1321

**Comment by fig02** [comment]
pr looks good to me after a couple pending comments are addressed

File: src/code/z_camera.c
Comment (engineer124) [reply src/code/z_camera.c]: It's worth noting that the are already 3 uses of `UNUSED` in z_camera.c, and that a lot of warnings are added without the `UNUSED`. But I'll leave it to that PR to sort out, so I've removed them

File: include/ultra64/xstdio.h
Comment (Dragorn421) [inline include/ultra64/xstdio.h:35]: this change shouldn't be there once you merge main (or run formatter) then we're good to go

**Pull request merged**
